calyx-py/calyx/builder.py

```diff
@@ -71,6 +72,14 @@ def __init__(
         for cell in cells:
             self.index[cell.id.name] = CellBuilder(cell)
         self.continuous = GroupBuilder(None, self)
+        random.seed(4)
+
+    def generate_name(self, prefix: str) -> str:
+        """Generate a unique name with the given prefix."""
+        while True:
+            name = f"{prefix}_{random.randint(0, 2**32)}"
+            if name not in self.index:
+                return name
```

Comment:
Here's one way to avoid randomness: just use an increasing counter. That is, keep this loop around, but instead of generating a new `randint` each time, bump a counter to the next value. Call it `next_gen_idx` or something, and start it at zero in `__init__`, and do `self.next_gen_idx += 1` inside the loop.

It's roughly the same thing but will just make stuff look more predictable: it will generate `eq_0`, `lt_1`, `eq_2` …, instead of `eq_83219`, `lt_39842340`, `eq_239`, ….

Reply:
Nice, thanks!